cli/scorecard/score.go

```diff
@@ -148,19 +150,67 @@ func (inventory *InventoryConfig) Score(config *ScoringConfig) error {
 	err = config.attachViolations(auditResult)
 
 	if len(auditResult.Violations) > 0 {
-		fmt.Printf("\n\n%v total issues found\n", len(auditResult.Violations))
-		for _, category := range config.categories {
-			fmt.Printf("\n\n%v: %v issues found\n", category.Name, category.Count())
-			fmt.Printf("----------\n")
-			for _, cv := range category.constraints {
-				fmt.Printf("%v: %v issues\n", cv.GetName(), cv.Count())
-				for _, v := range cv.Violations {
-					fmt.Printf("- %v\n\n",
-						v.Message,
-					)
-					Log.Debug("Violation metadata", "metadata", v.GetMetadata())
+		content := ""
+		switch outputFormat {
+		case "json":
+			type violationOutput struct {
+				Category   string
+				Constraint string
+				Resource   string
+				Message    string
+			}
+			var vOutput violationOutput
+			for _, category := range config.categories {
+				for _, cv := range category.constraints {
+					for _, v := range cv.Violations {
+						vOutput.Category = category.Name
+						vOutput.Constraint = v.Constraint
+						vOutput.Resource = v.Resource
+						vOutput.Message = v.Message
+						byteContent, err := json.MarshalIndent(vOutput, "", "  ")
+						if err != nil {
+							return err
+						}
+						content += string(byteContent) + "\n"
+						Log.Debug("Violation metadata", "metadata", v.GetMetadata())
```

Comment:
Shouldn't we simply include metadata in the JSON output?

Reply:
included in 372a085